### PR TITLE
Feature/profileforms

### PR DESCRIPTION
**Pull request by josegocampo**
Hi guys, sorry for all the delays and problems on my side.

I set up the Reducers, Actions and their flows with Redux for the User Profile Data and for the User Id.

**TLDR version:**  I have the Profiles working, you can access them in the Profile button of the Nav Bar, I am using Okta's data for now because I don't have access to the real API, but everything is working as intended and adding the API calls to use the real data shouldn't be more than adding a few lines of code to the corresponding Actions. 

Also styles are kinda messy but they are just a mock version for testing stuff out.

**Detailed Version:**

I added the Profiles and most things related to them, you can access them on the Nav Bar that Sohpia built.

For the User Profile Data right now I am using the userData that Okta provides for each individual user login that they gave us on https://docs.labs.lambdaschool.com/guides/okta/okta-basics. I am grabbing that from the useOktaAuth hook and then sending it to Redux's flow. 

Same thing for the User Id, which is not really being used yet, because I don't have the API working with Docker or Heroku, so I'm still saving it for when we switch the data from Okta's users mock data I am using to the real API data, so then we can make the API call and use that User Id as the /:userid mentioned in the profiles endpoint for each individual user.

So basically the end idea would be, grabbing the userId form the **sub** key value pair that the Okta hook gives, passing that trough the reducer and saving it to global state so then we can make an API call in the postProfileReducer to our axiosWithAuth().get(/**userId**) and save that to global state with the before mentioned postProfileReducer.

Everything with the Redux flow is working, but I am not making the API calls, which is just a matter of adding them inside the corresponding actions, so once we have the API ready it shouldn't be a problem adding them.

There are some inline JSX styles added which I don't intend to keep permanently but I was using to test stuff quickly.

The Whole styling of the Profiles is open to changes too, it is just a mock of what the final version would be.

I also implemented inside each Profile the **Edit Form** for the user profile, which is working with Redux and the Okta data we are using, so it is not how the final version would go, because Okta's data and the one in our API is different, but its there as a place holder. 

Inside each User Profile I also have a **button which is for manually switching between the views of the 2 kinds of user types,** Groomer or Client, because I don't have access yet to the user types from the API and Okta doesn't discriminate between user types, so its another manual placer holder but that should be really easy to implement once we have the real user data, everything is ready to receive the real data and switch some lines around and it should be working properly. You can see in the ProfilePage page and specifically the ProgilePageContainer.js that I added some local state to change the user type around. 

I implemented Immer's produce to the Reducers, which should be pretty self explanatory and intuitive as to why they are better to read and understand than the normal spreading of state and then modifying it, if anyone needs further explanation as to how they work please let me know.

**Next Steps:** 

- Adding the real API calls
- Reworking Styles
- Changing Data displayed to match the real API data
- Adding the Shop/Location Data to the Groomers Profile
- Adding the Favorite Groomers Data to the Clients Profile

Please let me know anything that is confusing or can be improved, I'm sure I'm missing some console.logs and there is potential for better abstractions in certain places, but I have to push something before this gets too big.

### CODE REVIEW

File: src/components/pages/ProfilePage/ClientProfilePage.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import { ProfileEditForm } from '../ProfileEdit';
+import profilepic from './profilepic.png';
+
+const ClientProfilePage = ({ user }) => {
+  const [profileEdit, setProfileEdit] = useState(false);
+
+  const handleChanges = e => {
+    setProfileEdit(true);
+  };
+  console.log(user);
```
Comment (dvwhite) [inline src/components/pages/ProfilePage/ClientProfilePage.js]: We should remove console logs from the components

File: src/components/pages/ProfilePage/ProfilePageContainer.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import GroomerProfilePage from './GroomerProfilePage';
+import ClientProfilePage from './ClientProfilePage';
+
+const ProfilePageContainer = ({ userData }) => {
+  const [userId, setUserId] = useState(1);
+
+  const handleChange = () => {
```
Comment (dvwhite) [inline src/components/pages/ProfilePage/ProfilePageContainer.js:9]: Is this for working with mock data?

File: src/components/pages/Home/HomeContainer.js [hunk 3]
```diff
@@ -15,29 +21,33 @@ function HomeContainer({ LoadingComponent }) {
...
         if (isSubscribed) {
-          setUserInfo(info);
+          postUserId(info.sub);
+          postProfile(info);
         }
       })
       .catch(err => {
         isSubscribed = false;
-        return setUserInfo(null);
+        console.log(err);
```
Comment (dvwhite) [inline src/components/pages/Home/HomeContainer.js]: We should remove console logs from the components

**Review by dvwhite — CHANGES_REQUESTED** [review]
Once tests no longer fail and the console logs are converted to Harry's dev-only `log` function then it will pass review

File: src/state/index.js [hunk 4]
```diff
@@ -1,9 +1,17 @@
...
-// STORE
-const store = createStore(reducer, composeWithDevTools(applyMiddleware(thunk)));
+const middleWare = [thunk];
 
-export default store;
+export default createStore(rootReducer, applyMiddleware(...middleWare));
+
+// import { createStore, applyMiddleware } from 'redux';
+// import { composeWithDevTools } from 'redux-devtools-extension';
+// import { reducer } from './reducers';
```
Comment (dvwhite) [inline src/state/index.js]: Commented code should be removed (especially if not used).

File: src/test-utils/test-utils.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+// Import your own reducer
+import configureStore from 'redux-mock-store';
+import reducer from '../state/reducers/index';
+import thunk from 'redux-thunk';
+
+// The mocked redux store
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+
+function render(
```
Comment (hmerritt) [inline src/test-utils/test-utils.js:15]: The custom render function is a great idea!

File: src/components/pages/ProfilePage/ClientProfilePage.js (hunk 1 above)
Comment (josegocampo) [reply src/components/pages/ProfilePage/ClientProfilePage.js]: Done

File: src/state/index.js (hunk 4 above)
Comment (josegocampo) [reply src/state/index.js]: Done!

File: src/components/pages/Home/HomeContainer.js (hunk 3 above)
Comment (josegocampo) [reply src/components/pages/Home/HomeContainer.js]: Done!

File: src/components/pages/ProfilePage/ProfilePageContainer.js (hunk 2 above)
Comment (josegocampo) [reply src/components/pages/ProfilePage/ProfilePageContainer.js:9]: Yes, but updating today to real data

File: src/state/index.js [hunk 6]
```diff
@@ -5,13 +5,3 @@ import rootReducer from './reducers';
 const middleWare = [thunk];
 
 export default createStore(rootReducer, applyMiddleware(...middleWare));
```
Comment (dvwhite) [inline src/state/index.js:7]: Thank you for removing the commented code!

**Review by dvwhite — APPROVED** [review]
Tests are passing and all other changes requested made. Thanks, Jose!

File: src/state/reducers/UserIdReducer.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import produce from 'immer';
```
Comment (hmerritt) [inline src/state/reducers/UserIdReducer.js:1]: Using `immer` was a great idea - reducer looks much cleaner.

**Review by hmerritt — APPROVED** [review]
Great work on adding `immer` to the reducers and creating both profile pages.

**Pull request merged**